### PR TITLE
support tooltip in the charts using d3.tip

### PR DESCRIPTION
**Pull request by hackmod**
## screenshot
image

(Ethersocial(ESN) case with a dark theme)

### CODE REVIEW

File: public/js/controllers/StatsController.js [hunk 1]
```diff
@@ -795,21 +856,36 @@ angular.module('BlocksApp').controller('StatsController', function($stateParams,
...
+                    tip.show(s1, this);
+                    tip.style("left", d3.event.pageX + 10 + "px");
+                    tip.style("top", d3.event.pageY + 25 + "px");
+                    focus.style("display", null);
+                 })
+                .on("mouseout", function() { tip.hide(); focus.style("display", "none"); })
                 .on("mousemove", mousemove);
 
-            function mousemove() {
+            function mousemove(d) {
```
Comment (YazzyYaz) [inline public/js/controllers/StatsController.js]: is there any reason for adding the `d` function parameter? It's not being used anywhere in the function below, but `data` on line 881 is passed as a `d` parameter.

File: public/js/controllers/StatsController.js [hunk 2]
```diff
@@ -414,17 +424,27 @@ angular.module('BlocksApp').controller('StatsController', function($stateParams,
...
+                    tip.style("left", d3.event.pageX + 10 + "px");
+                    tip.style("top", d3.event.pageY + 25 + "px");
+                    focus.style("display", null);
+                 })
+                .on("mouseout", function() { tip.hide(); focus.style("display", "none"); })
                 .on("mousemove", mousemove);
 
 
-            function mousemove() {
+            function mousemove(d) {
```
Comment (YazzyYaz) [inline public/js/controllers/StatsController.js]: Same question as the one below. Why pass a `d` function parameter if it's not being used?

File: public/js/controllers/StatsController.js [hunk 3]
```diff
@@ -610,21 +644,36 @@ angular.module('BlocksApp').controller('StatsController', function($stateParams,
...
 
+            svg.call(tip);
+
             svg.append("rect")
                 .attr("class", "overlay")
                 .attr("width", width)
                 .attr("height", height)
-                .on("mouseover", function() { focus.style("display", null); })
-                .on("mouseout", function() { focus.style("display", "none"); })
+                .on("mouseover", function(d) {
```
Comment (YazzyYaz) [inline public/js/controllers/StatsController.js]: same comment about passing the `d` parameter. I notice in this example here: http://bl.ocks.org/d3noob/a22c42db65eb00d4e369 That they pass the `d` parameter, but they're using it inside the function. For example, 
```
.on("mouseover", function(d) {		
            div.transition()		
                .duration(200)		
                .style("opacity", .9);		
            div	.html(formatTime(d.date) + "<br/>"  + d.close)	
                .style("left", (d3.event.pageX) + "px")		
                .style("top", (d3.event.pageY - 28) + "px");	
            })		
```
Here, the `d` is being referenced inside the function in `d.date` and `d.close`

File: public/js/controllers/StatsController.js [hunk 4]
```diff
@@ -610,21 +644,36 @@ angular.module('BlocksApp').controller('StatsController', function($stateParams,
...
+                    tip.show(s1, this);
+                    tip.style("left", d3.event.pageX + 10 + "px");
+                    tip.style("top", d3.event.pageY + 25 + "px");
+                    focus.style("display", null);
+                 })
+                .on("mouseout", function() { tip.hide(); focus.style("display", "none"); })
                 .on("mousemove", mousemove);
 
-            function mousemove() {
+            function mousemove(d) {
```
Comment (YazzyYaz) [inline public/js/controllers/StatsController.js]: Same comment on `d` as before.

File: public/js/controllers/StatsController.js [hunk 5]
```diff
@@ -783,6 +832,18 @@ angular.module('BlocksApp').controller('StatsController', function($stateParams,
                 .attr("class", "y axis")
                 .call(yAxis);
 
+            var tip = d3.tip()
+                .attr('class', 'd3-tip')
+                .offset([-10, 0])
+                .direction('s')
+                .html(function(d) {
```
Comment (YazzyYaz) [inline public/js/controllers/StatsController.js]: This is a good example of when `d` function parameter is being used inside the function, as it's being used in line 843 in `d.unixtime`

File: public/js/controllers/StatsController.js [hunk 6]
```diff
@@ -795,21 +856,36 @@ angular.module('BlocksApp').controller('StatsController', function($stateParams,
...
 
+            svg.call(tip);
+
             svg.append("rect")
                 .attr("class", "overlay")
                 .attr("width", width)
                 .attr("height", height)
-                .on("mouseover", function() { focus.style("display", null); })
-                .on("mouseout", function() { focus.style("display", "none"); })
+                .on("mouseover", function(d) {
```
Comment (YazzyYaz) [inline public/js/controllers/StatsController.js]: Same comment on `d` function parameter as before.

Comment (hackmod) [reply public/js/controllers/StatsController.js]: thank you!

remove some needless arg

**Comment by hackmod** [comment]
fixed! thanks in advance!

File: public/js/controllers/StatsController.js (hunk 5 above)
Comment (hackmod) [reply public/js/controllers/StatsController.js]: thank you!

**Pull request merged**